src/worker.cpp

```diff
@@ -1185,4 +1184,284 @@ void Worker::collectFrusFromJson()
+    {
+        return m_parsedJson["backupAndRestore"].value("PrimaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+std::string Worker::getSecondaryHwPath()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("SecondaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+types::BackupMapVariant Worker::getBackupMap()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        const std::string& l_backupMap =
+            m_parsedJson["backupAndRestore"].value("BackupMap", "");
+        if (constants::VpdBackupMap.find(l_backupMap) !=
+            constants::VpdBackupMap.end())
+        {
+            return constants::VpdBackupMap.at(l_backupMap);
+        }
+    }
+    return std::monostate{};
+}
+
+void Worker::backupAndRestore(types::VPDMapVariant& io_parsedVpdMap)
+{
+    logging::logMessage("BackupAndRestore is started.");
+    const std::string& l_primaryHwPath = getPrimaryHwPath();
+    const std::string& l_secondaryHwPath = getSecondaryHwPath();
+
+    if (l_primaryHwPath.empty() && l_secondaryHwPath.empty())
+    {
+        logging::logMessage(
+            "Both primary and secondary path's are empty, can't initiate BackupAndRestore.");
+        return;
+    }
+
+    const auto& l_backupMap = getBackupMap();
+    if (std::holds_alternative<std::monostate>(l_backupMap))
+    {
+        logging::logMessage(
+            "Couldn't find system specific backup map, can't initiate BackupAndRestore.");
+        return;
+    }
+
+    types::VPDMapVariant l_parsedPrimaryVpdMap;
+    if (!l_primaryHwPath.empty())
+    {
+        fillVPDMap(l_primaryHwPath, l_parsedPrimaryVpdMap);
+    }
+
+    types::VPDMapVariant l_parsedSecVpdMap;
+    if (!l_secondaryHwPath.empty())
+    {
+        fillVPDMap(l_secondaryHwPath, l_parsedSecVpdMap);
+    }
+
+    if (!l_primaryHwPath.empty() && !l_secondaryHwPath.empty() &&
+        l_parsedPrimaryVpdMap.index() != l_parsedSecVpdMap.index())
+    {
+        logging::logMessage(
+            "Primary and secondary VPD types are not matching.");
+        return;
+    }
+
+    // Implement backup and restore for ipz type VPD
+    if ((std::holds_alternative<types::IPZVpdMap>(l_parsedPrimaryVpdMap) ||
+         std::holds_alternative<types::IPZVpdMap>(l_parsedSecVpdMap)) &&
+        std::holds_alternative<types::IBMIpzBackupMap>(l_backupMap))
+    {
+        backupAndRestoreIpzVpd(io_parsedVpdMap, l_backupMap,
+                               l_parsedPrimaryVpdMap, l_parsedSecVpdMap);
+    }
+    else if (std::holds_alternative<types::KeywordVpdMap>(
+                 l_parsedPrimaryVpdMap) ||
+             std::holds_alternative<types::KeywordVpdMap>(l_parsedSecVpdMap))
+    {
+        // To Do, handle implementation for Keyword type VPD
+    }
+}
+
+void Worker::backupAndRestoreIpzVpd(
```

Comment:
This is a concrete implementation for IPZ. Why this is taking variant as a parameter? Need not allow caller to call this API with irrelevant type.
Please fix the API.

Reply:
Updated to IPZ type.